lib/gyro/liquidgen/generator.rb

```diff
@@ -24,81 +24,56 @@ module Liquidgen
     #
     class Generator
 
-      attr_accessor :root_template, :filename_template, :params, :output_dir
-      attr_accessor :enum_filename_template, :enum_filename_template_path, :enum_template_path, :enum_template
+      attr_accessor :params, :output_dir, :template_dir
 
       # PUBLIC METHODS #######################################################
       def initialize(template_dir, output_dir, params)
         Gyro::Log.title('Generating Model')
 
         @params = params
         @output_dir = output_dir
-
-        # Parse object template name
-        root_template_path = template_dir + 'root.liquid'
-        Gyro::Error.exit_with_error('Bad template directory content ! Your template need to include root.liquid file') unless root_template_path.exist?
-        root_template_string = root_template_path.read
-        @root_template = Liquid::Template.parse(root_template_string)
+        @template_dir = template_dir
 
         # Define Template path for Liquid file system to use Include Tag
         Liquid::Template.file_system = Liquid::LocalFileSystem.new(template_dir)
-        # Parse object template
-        filename_template_path = (template_dir + 'filename.liquid')
-        Gyro::Error.exit_with_error('Bad template directory content ! Your template need to include filename.liquid file') unless filename_template_path.exist?
-        filename_template_string = filename_template_path.readlines.first
-        @filename_template = Liquid::Template.parse(filename_template_string)
-
-        # Parse enum template
-        @enum_template_path = (template_dir + 'enum.liquid')
-        
-        enum_template_string = enum_template_path.read
-        @enum_template = Liquid::Template.parse(enum_template_string)
-        
-        # Parse enum template name
-        @enum_filename_template_path = (template_dir + 'enum_filename.liquid').exist? ? (template_dir + 'enum_filename.liquid') : (template_dir + 'filename.liquid')
-        enum_filename_template_string = enum_filename_template_path.readlines.first
-        @enum_filename_template = Liquid::Template.parse(enum_filename_template_string)
+      end
 
+      def generate(xcdatamodel)
+        generate_root_template(xcdatamodel)
         Gyro::Log.success('Model objects are generated !')
       end
 
-      def generate(xcdatamodel)
+      private ################################################################
 
+      def generate_root_template(xcdatamodel) 
```

Comment:
I'd rather rename that `generate_entities_files` or `generate_entities_from_root_template`, as it generates files from templates, it doesn't generate the root template itself.